Neos.ContentGraph.DoctrineDbalAdapter/src/Domain/Repository/ContentGraph.php

```diff
@@ -188,6 +189,42 @@ public function findParentNodeAggregates(
         return $this->mapQueryBuilderToNodeAggregates($queryBuilder);
     }
 
+    public function findAncestorNodeAggregateIds(NodeAggregateId $entryNodeAggregateId): NodeAggregateIds
+    {
+        $queryBuilderInitial = $this->createQueryBuilder()
+            ->select('n.nodeAggregateId, ph.parentnodeanchor')
+            ->from($this->nodeQueryBuilder->tableNames->node(), 'n')
+            // we need to join with the hierarchy relation, because we need the node name.
+            ->innerJoin('n', $this->nodeQueryBuilder->tableNames->hierarchyRelation(), 'ch', 'ch.parentnodeanchor = n.relationanchorpoint')
+            ->innerJoin('ch', $this->nodeQueryBuilder->tableNames->node(), 'c', 'c.relationanchorpoint = ch.childnodeanchor')
+            ->innerJoin('n', $this->nodeQueryBuilder->tableNames->hierarchyRelation(), 'ph', 'n.relationanchorpoint = ph.childnodeanchor')
```

Comment:
This query again joins, the hierachy relation twice. I think this is not needed. See: https://github.com/neos/neos-development-collection/pull/5268

Reply:
has been adjusted